src/main/java/org/opentripplanner/routing/api/request/RoutingRequest.java

```diff
@@ -1264,40 +1236,20 @@ public void setRoutingContext(Graph graph) {
      * TODO rename - this is not a "setter", it creates a new routingContext, which has side effects on Graph
      *               (Constructors with side effects on their parameters are a bad design).
      */
-    public void setRoutingContext(Graph graph, Edge fromBackEdge, Vertex from, Vertex to) {
```

Comment:
How about resolving the TODO above and calling the method something like `copyWithRoutingContext` or similar?

Reply:
I want to get rid of routing context altogether as a field in the Routing request, so this method will eventually be removed.

Reply:
If you rename the method, the comment can be deleted as well. It do point to another problem, but this is well known and the comment belong in the constructor not here.